### PR TITLE
TestDiscovery_ConnectedPopulatesRoutingTable fix

### PR DESCRIPTION
**Pull request by igorcrevar**
# Description

Just a fix for one unit test: The routing table is filled after peers connect to each other (and therefore populate their peers map). So, it is not enough to wait for the peers map to be populated.

# Changes include

- [X] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [X] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [X] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

**Review by goran-ethernal — APPROVED** [review]

File: network/discovery_e2e_test.go [hunk 1]
```diff
@@ -31,8 +34,14 @@ func TestDiscovery_ConnectedPopulatesRoutingTable(t *testing.T) {
...
 	}
 
-	assert.Equal(t, servers[0].discovery.RoutingTableSize(), 1)
-	assert.Equal(t, servers[1].discovery.RoutingTableSize(), 1)
+	// make sure each routing table has peer
+	if _, err := WaitUntilRoutingTableToBeFilled(ctx, servers[0], 1); err != nil {
+		t.Fatalf("server 0 should add a peer to routing table but didn't, peer=%s", servers[1].host.ID())
+	}
+
+	if _, err := WaitUntilRoutingTableToBeFilled(ctx, servers[1], 1); err != nil {
```
Comment (Stefan-Ethernal) [inline network/discovery_e2e_test.go]: This is not directly related to the PR, but could you consider renaming `WaitUntilRoutingTableToBeFilled` to `WaitUntilRoutingTableIsFilled`?

File: network/discovery_e2e_test.go [hunk 2]
```diff
@@ -31,8 +34,14 @@ func TestDiscovery_ConnectedPopulatesRoutingTable(t *testing.T) {
 		t.Fatalf("Unable to join peers, %v", joinErr)
 	}
 
-	assert.Equal(t, servers[0].discovery.RoutingTableSize(), 1)
-	assert.Equal(t, servers[1].discovery.RoutingTableSize(), 1)
+	// make sure each routing table has peer
+	if _, err := WaitUntilRoutingTableToBeFilled(ctx, servers[0], 1); err != nil {
+		t.Fatalf("server 0 should add a peer to routing table but didn't, peer=%s", servers[1].host.ID())
```
Comment (Stefan-Ethernal) [inline network/discovery_e2e_test.go]: You didn't want to put `require.NoError` instead and in that case we can avoid if.

**Review by Stefan-Ethernal — APPROVED** [review]
Leaving a couple of optional notes to consider.

File: network/discovery_e2e_test.go (hunk 2 above)
Comment (igorcrevar) [reply network/discovery_e2e_test.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1745/commits/4e54b84faaea7e7058609da315771e40243255be

File: network/discovery_e2e_test.go (hunk 1 above)
Comment (igorcrevar) [reply network/discovery_e2e_test.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1745/commits/4e54b84faaea7e7058609da315771e40243255be
